sdk/java-sdk-protobuf-testkit/src/main/java/kalix/javasdk/testkit/EventingTestKit.java

```diff
@@ -63,44 +96,56 @@ interface Topic {
      * @param timeout amount of time to wait for a message if it was not received already
      * @return message including ByteString payload and metadata
      */
-    Message<ByteString> expectOne(Duration timeout);
+    Message<Object> expectOne(Duration timeout);
 
     /**
      * Waits and returns the next unread message on this topic and automatically parses
      * and casts it to the specified given type.
      *
-     * @param instance object to be used to parse the received message bytes
+     * @param instance class type to cast the received message bytes to
      * @return a Message of type T
      * @param <T> a given domain type
      */
-    <T extends GeneratedMessageV3> Message<T> expectOneClassOf(T instance);
+    <T extends GeneratedMessageV3> Message<T> expectMessageType(Class<T> instance);
+
+    /**
+     * Waits and returns the next unread message on this topic and automatically parses
+     * and casts it to the specified given type.
+     * Note the message might have been received before this method was called.
+     * If no message is received, a timeout exception is thrown.
+     *
+     * @param timeout amount of time to wait for a message if it was not received already
+     * @return message including ByteString payload and metadata
+     */
+    <T extends GeneratedMessageV3> Message<T> expectMessageType(Class<T> instance, Duration timeout);
 
     /**
      * Waits for a default amount of time before returning all unread messages in the topic.
      * If no message is received, a timeout exception is thrown.
      *
-     * @return collection of messages, each message including ByteString payload and metadata
+     * @return collection of messages, each message including the deserialized payload object and metadata
      */
-    Collection<Message<ByteString>> expectN();
+    Collection<Message<Object>> expectN();
```

Comment:
Could it be `Message<?>` because we don't know what types they are? (I don't completely remember the semantics of Java wildcards), and `Message[_]` for the Scala ones

Reply:
fixed now in https://github.com/lightbend/kalix-jvm-sdk/pull/1586/commits/e8deab4b95871ed04fd9c1a76cdac0deca1ee3f9